x-pack/metricbeat/module/statsd/server/data.go

```diff
@@ -33,20 +33,32 @@ type statsdMetric struct {
 
 func splitTags(rawTags, kvSep []byte) map[string]string {
 	tags := map[string]string{}
-	for _, kv := range bytes.Split(rawTags, []byte(",")) {
-		kvSplit := bytes.SplitN(kv, kvSep, 2)
-		if len(kvSplit) != 2 {
-			logger.Warn("could not parse tags")
-			continue
+	if bytes.Contains(rawTags, []byte(",")) {
+		for _, kv := range bytes.Split(rawTags, []byte(",")) {
+			kvSplit := bytes.SplitN(kv, kvSep, 2)
+			if len(kvSplit) != 2 {
+				logger.Warn("could not parse tags")
+				continue
+			}
+			tags[string(kvSplit[0])] = string(kvSplit[1])
+		}
+	} else {
+		for _, kv := range bytes.Split(rawTags, []byte(";")) {
+			kvSplit := bytes.SplitN(kv, kvSep, 2)
+			if len(kvSplit) != 2 {
+				logger.Warn("could not parse tags")
+				continue
+			}
+			tags[string(kvSplit[0])] = string(kvSplit[1])
```

Comment:
We can reduce the code repetition like this:

```go
func splitTags(rawTags, kvSep []byte) map[string]string {
    tags := map[string]string{}
    var tagSplit [][]byte

    if bytes.Contains(rawTags, []byte(",")) {
        tagSplit = bytes.Split(rawTags, []byte(","))
    } else { // or use else if??
        tagSplit = bytes.Split(rawTags, []byte(";"))
    }

    for _, kv := range tagSplit {
        kvSplit := bytes.SplitN(kv, kvSep, 2)
        if len(kvSplit) != 2 {
            logger.Warn("could not parse tags")
            continue
        }
        tags[string(kvSplit[0])] = string(kvSplit[1])
    }

    return tags
}
``` 


I am just thinking about something. Is it possible to have an escaped separator in the tag's key/value? If yes, then should we handle? Maybe not as part of this PR. Code change is simple.

In the loop, we just have to do:
```
kv = bytes.ReplaceAll(kv, []byte("\\,"), []byte(","))
kv = bytes.ReplaceAll(kv, []byte("\\;"), []byte(";"))
kv = bytes.ReplaceAll(kv, []byte("\\:"), []byte(":"))
```

Reply:
The only example I can find or test for myself is if the metric name has `:` as seen [here](https://github.com/statsd/statsd/issues/585)

Reply:
Yeah, perhaps we should do this enhancement in a later PR. It immediately hit me and thought the logic should be there.

Anyway, for now, you can avoid repetition in code by using the snippet I gave above.